src/main/java/grandduke/command/Parser.java

```diff
@@ -0,0 +1,203 @@
+     *              the input from the user
+     * @param type
+     *              the type of task
+     * @return the new task
+     */
+    public static Task parseNewTask(String input, String type) throws GrandException {
+        Task newTask;
+
+        try {
+            switch (type) {
+                case Io.TODO_COMMAND:
+                    newTask = createTodo(input);
+                    break;
+                case Io.DEADLINE_COMMAND:
+                    newTask = createDeadline(input);
+                    break;
+                case Io.EVENT_COMMAND:
+                    newTask = createEvent(input);
+                    break;
+                default:
+                    newTask = null;
+                    break;
+            }
+        } catch (GrandException e) {
+            throw e;
+        }
+
+        return newTask;
+    }
+
+    /**
+     * Creates a new Todo task
+     * 
+     * @param input
+     *              the input from the user
+     * @return the new Todo task
+     */
+    public static Task createTodo(String input) throws EmptyTodoException {
+        if (input.equals("")) {
+            throw new EmptyTodoException();
+        }
+        return new Todo(input);
+    }
+
+    /**
+     * Creates a new Deadline task
+     * 
+     * @param input
+     *              the input from the user
+     * @return the new Deadline task
+     */
+    public static Task createDeadline(String input) throws EmptyDeadlineException, MissingByException,
+            EmptyDeadlineDescriptionException, EmptyDeadlineDateException {
+
+        if (input.equals("")) {
+            throw new EmptyDeadlineException();
+        }
+
+        if (!input.contains("/by")) {
+            throw new MissingByException();
+        }
+
+        String[] inputList = input.split("/by", 2);
+        String taskDesc = inputList[0].strip();
+        if (taskDesc.equals("")) {
+            throw new EmptyDeadlineDescriptionException();
+        }
+
+        String deadline = inputList[1].strip();
+        if (deadline.equals("")) {
+            throw new EmptyDeadlineDateException();
+        }
+
+        return new Deadline(taskDesc, deadline);
+    }
+
+    /**
+     * Creates a new Event task
+     * 
+     * @param input
+     *              the input from the user
+     * @return the new Event task
+     */
+    public static Task createEvent(String input) throws EmptyEventException, MissingFromException,
+            MissingToException, EmptyEventDescriptionException, EmptyEventFromException,
+            EmptyEventToException {
+
+        if (input.equals("")) {
+            throw new EmptyEventException();
+        }
+
+        if (!input.contains("/from")) {
+            throw new MissingFromException();
+        }
+
+        if (!input.contains("/to")) {
+            throw new MissingToException();
+        }
+
+        String[] inputList = input.split("/from", 2);
+        String[] inputList2 = inputList[1].split("/to", 2);
```

Comment:
For better naming convention, you are encouraged not to use names like "inputList1", "inputList2". Instead, try to use something more meaningful and intuitive to read such as "inputList" and "inputListFinal".